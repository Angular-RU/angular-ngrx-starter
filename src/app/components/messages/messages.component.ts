import { Component, OnInit } from '@angular/core';
import { MessageService } from '@appServices/message.service';

@Component({
  selector: 'app-messages',
  templateUrl: './messages.component.html',
  styleUrls: ['./messages.component.css']
})
export class MessagesComponent {
  constructor(public messageService: MessageService) {}
}
