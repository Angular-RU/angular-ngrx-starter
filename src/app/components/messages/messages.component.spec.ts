import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { MessageService } from '@appServices/message.service';
import { MessagesComponent } from './messages.component';
import { mock, instance } from 'ts-mockito';

describe('MessagesComponent', () => {
  let comp: MessagesComponent;
  let fixture: ComponentFixture<MessagesComponent>;
  let messageServiceMock: MessageService;

  beforeEach(() => {
    messageServiceMock = mock(MessageService);

    TestBed.configureTestingModule({
      declarations: [MessagesComponent],
      schemas: [NO_ERRORS_SCHEMA],
      providers: [
        {
          provide: MessageService,
          useFactory: () => instance(messageServiceMock)
        }
      ]
    });
    fixture = TestBed.createComponent(MessagesComponent);
    comp = fixture.componentInstance;
  });

  it('can load instance', () => {
    expect(comp).toBeTruthy();
  });

  it('should show messages', () => {
    const service = fixture.debugElement.injector.get(MessageService);
    service.messages = ['test message', 'test message 2'];
    fixture.detectChanges();

    const html = fixture.debugElement.nativeElement;

    expect(html).toMatchSnapshot();
  });
});
