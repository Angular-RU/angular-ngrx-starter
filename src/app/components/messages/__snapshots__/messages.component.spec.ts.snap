// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesComponent should show messages 1`] = `
<div
  id="root1"
  ng-version="7.1.4"
>
  
  <div>
    <h2>
      Messages
    </h2>
    <button
      class="clear"
    >
      clear
    </button>
    
    <div>
       test message 
    </div>
    <div>
       test message 2 
    </div>
  </div>
</div>
`;
